build.zig

```diff
@@ -930,6 +1089,267 @@ test hex64 {
+        "--vendor-insts-output", extinst_file,
+        "--vendor-operand-kind-prefix", operand_kind_prefix,
+    };
+
+    exec(allocator, args, sdkPath("/")) catch |err| 
+    {
+        log.err("Failed to generate SPIR-V vendor table '" ++ name ++ "'. error: {s}", .{@errorName(err)});
+        std.process.exit(1);
+    };
+}
+
+fn genSPIRVRegistryTables(allocator: std.mem.Allocator) void {
+    const script = spirv_tools_path ++ "/utils/generate_registry_tables.py";
+
+    const xml_file = spirv_headers_path ++ "/include/spirv/spir-v.xml";
+    const inc_file = spirv_output_path ++ "/generators.inc";
+
+    const args = &[_][]const u8 {
+        "python3", script,
+        "--xml", xml_file,
+        "--generator", inc_file,
+    };
+
+    exec(allocator, args, sdkPath("/")) catch |err| 
+    {
+        log.err("Failed to generate SPIR-V registry tables. error: {s}", .{@errorName(err)});
+        std.process.exit(1);
+    };
+}
+
+fn buildSPIRVVersion(allocator: std.mem.Allocator) void {
+    const script = spirv_tools_path ++ "/utils/update_build_version.py";
+
+    const changes_file = spirv_tools_path ++ "/CHANGES";
+    const inc_file = spirv_output_path ++ "/build-version.inc";
+
+    const args = &[_][]const u8 {
+        "python3", script,
+        changes_file,
+        inc_file,
+    };
+
+    exec(allocator, args, sdkPath("/")) catch |err| 
+    {
+        log.err("Failed to generate SPIR-V build version. error: {s}", .{@errorName(err)});
+        std.process.exit(1);
+    };
+}
+
+fn generateSPIRVGrammar(allocator: std.mem.Allocator) void {
+    ensurePython(allocator);
+
+    genSPIRVCoreTables(allocator, "unified1");
+    genSPIRVCoreEnums(allocator, "unified1");
+
+    genSPIRVGlslTables(allocator, "1.0");
+
+    genSPIRVOpenCLTables(allocator, "1.0");
+
+    genSPIRVLanguageHeader(allocator, "DebugInfo", spvHeaderFile("unified1", "extinst.debuginfo.grammar.json"));
+    genSPIRVLanguageHeader(allocator, "OpenCLDebugInfo100", spvHeaderFile("unified1", "extinst.opencl.debuginfo.100.grammar.json"));
+    genSPIRVLanguageHeader(allocator, "NonSemanticShaderDebugInfo100", spvHeaderFile("unified1", "extinst.nonsemantic.shader.debuginfo.100.grammar.json"));
+
+    genSPIRVVendorTable(allocator, "spv-amd-shader-explicit-vertex-parameter", "...nil...");
+    genSPIRVVendorTable(allocator, "spv-amd-shader-trinary-minmax", "...nil...");
+    genSPIRVVendorTable(allocator, "spv-amd-gcn-shader", "...nil...");
+    genSPIRVVendorTable(allocator, "spv-amd-shader-ballot", "...nil...");
+    genSPIRVVendorTable(allocator, "debuginfo", "...nil...");
+    genSPIRVVendorTable(allocator, "opencl.debuginfo.100", "CLDEBUG100_");
+    genSPIRVVendorTable(allocator, "nonsemantic.clspvreflection", "...nil...");
+    genSPIRVVendorTable(allocator, "nonsemantic.shader.debuginfo.100", "SHDEBUG100_");
+
+    genSPIRVRegistryTables(allocator);
+
+    buildSPIRVVersion(allocator);
+}
+
+const BuildSPIRVGrammarStep = struct {
```

Comment:
I have a goal to move away from having any custom build steps like this; so this will be a no-go

Reply:
No problem, this was only meant to generate spirv-headers, so if those are pregenerated it should be unneccesary.